### PR TITLE
Drive database defaults programmatically in the startup script

### PR DESCRIPTION
**Pull request by ibacher**
This allows multiple database systems to be properly supported by overriding a single property. As an example, I added the basic configuration for PostgreSQL. While we technically don't support PostgreSQL (but hopefully soon will), I thought this was useful for generalizing which properties we should drive from default property values.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,20 +36,53 @@ The below is a list of supported enviroment variables, along with their default
...
+* OMRS_CONFIG_CONNECTION_TYPE: ```mysql```
 * OMRS_CONFIG_CONNECTION_DRIVER_CLASS: ```com.mysql.jdbc.Driver```
 * OMRS_CONFIG_CONNECTION_USERNAME: ```openmrs```
 * OMRS_CONFIG_CONNECTION_PASSWORD: ```openmrs```
-* OMRS_CONFIG_CONNECTION_SERVER: ```db```
-* OMRS_CONFIG_CONNECTION_PORT: ```3306```
-* OMRS_CONFIG_CONNECTION_NAME: ```openmrs```
-* OMRS_CONFIG_CONNECTION_ARGS: ```autoReconnect=true&sessionVariables=storage_engine=InnoDB&useUnicode=true&characterEncoding=UTF-8```
-* OMRS_CONFIG_CONNECTION_EXTRA_ARGS: ``````
+* OMRS_CONFIG_CONNECTION_SERVER: ```localhost```
```
[the remark this answers is not in the document]
Comment (ibacher) [reply README.md:43]: So my thought here in using `localhost` was that `localhost` is the most sensible default for a pure Docker image. I agree that in the context of deploying this image in a Docker Compose stack, `localhost` is unlikely to be the right choice. This does put some onus on someone setting up the Docker Compose files, but I thought we’d probably be releasing reference Docker Compose stacks as well.

File: README.md [hunk 2]
```diff
@@ -36,20 +36,53 @@ The below is a list of supported enviroment variables, along with their default
...
+* OMRS_CONFIG_CONNECTION_DATABASE: ```openmrs```
```
[the remark this answers is not in the document]
Comment (ibacher) [reply README.md:44]: The property on the JDBC driver is `databaseName`, which is what prompted this, but I can revert it.

File: README.md [hunk 3]
```diff
@@ -36,20 +36,53 @@ The below is a list of supported enviroment variables, along with their default
...
 * OMRS_CONFIG_AUTO_UPDATE_DATABASE: ```false```
+* OMRS_CONFIG_CONNECTION_TYPE: ```mysql```
 * OMRS_CONFIG_CONNECTION_DRIVER_CLASS: ```com.mysql.jdbc.Driver```
 * OMRS_CONFIG_CONNECTION_USERNAME: ```openmrs```
 * OMRS_CONFIG_CONNECTION_PASSWORD: ```openmrs```
-* OMRS_CONFIG_CONNECTION_SERVER: ```db```
-* OMRS_CONFIG_CONNECTION_PORT: ```3306```
-* OMRS_CONFIG_CONNECTION_NAME: ```openmrs```
-* OMRS_CONFIG_CONNECTION_ARGS: ```autoReconnect=true&sessionVariables=storage_engine=InnoDB&useUnicode=true&characterEncoding=UTF-8```
-* OMRS_CONFIG_CONNECTION_EXTRA_ARGS: ``````
```
[the remark this answers is not in the document]
Comment (ibacher) [reply README.md:46]: I’ll add this back in. That’s a very good point.

**Comment by mseaton** [comment]
Thanks @ibacher - @cioan can you please have a look at our usage of this image on our test servers and determine if we need to update any of the environment variables set in our docker-compose or .env files, given some of the default values are changed or removed in this commit?

**Comment by cioan** [comment]
Thanks @ibacher . This looks good to me. Here is the docker config we use for the Rwanda test servers:
https://github.com/PIH/rwandaemr-installer/blob/master/src/main/resources/docker/docker-compose.yml

**Pull request merged**